CHANGELOG.md

```diff
@@ -1,10 +1,64 @@
-# 0.7.0 alpha.1
-
-Currently the changelog for v0.7.0 is incomplete, this tracked in issue #1217.
+# 0.7.0
+
+Version 0.7 of Mio contains various major changes compared to version 0.6.
+Overall a large number of API changes have been made to reduce the complexity of
+the implementation and remove overhead where possible.
+
+Also the upgrade guide in docs/upgrade_guide.md.
+
+## Added
+
+* New `Interest` structure that replaces `Ready` in registering event sources.
+* `Registry` structure was added to separate registering and polling.
+* `Waker` was added to allow another thread to wake a thread polling `Poll`.
+* Added Unix Domain Socket (UDS) types: `UnixDatagram`, `UnixListener` and
+  `UnixStream`.
+
+## Removed
+
+* All code deprecated in 0.6 was removed in 0.7.
+* Support for Fuchsia was removed, the code was unmaintained.
+* Support for Bitrig was removed, rustc dropped support for it also.
+* `UnixReady` was merged into `Ready`.
+* Custom user-space readiness queue was removed, this includes the public
+  `Registration` and `SetReadiness` types.
+* `PollOpt` was removed, now all registrations use edge-triggers. See the
+  upgrade guide on how to process event using edge-triggers.
+* The network types (types in the `net` module) now support only the same API as
+  found in the standard library, various methods on the types were removed.
+* `TcpStream` now supports vectored I/O.
+* `Poll::poll_interruptible` was removed and instead `Poll::poll` doesn't hide
```

Comment:
This is a little confusing to read. How about: `Poll::poll_interruptible is no longer provided. Poll::poll will now return an error if one occurs.`